File: examples/python/documentation/assets.py
# DOCS_BLOCK_START:assets
from constructs import Construct
from cdktf import TerraformAsset, TerraformStack, AssetType
from imports.aws.provider import AwsProvider
from imports.aws.s3_bucket import S3Bucket
from imports.aws.s3_bucket_object import S3BucketObject
import os


class MyAssetStack(TerraformStack):
    def __init__(self, scope: Construct, name: str):
        super().__init__(scope, name)

        AwsProvider(self, 'aws', region='eu-central-1')

        bucket = S3Bucket(self, "bucket", bucket="demo")

        asset = TerraformAsset(self, "lambda-asset",
                                     path=os.path.join(os.path.dirname(__file__), 'lambda'),
                                     type=AssetType.ARCHIVE
                                     )

        S3BucketObject(self, "lambda-archive",
                                            bucket=bucket.bucket,
                                            key=asset.file_name,
                                            source=asset.path
                                            )

# DOCS_BLOCK_END:assets

'''
DOCS_BLOCK_START:assets
app = cdktf.App()
MyAssetStack(app, "demo")
app.synth()
DOCS_BLOCK_END:assets
'''

File: examples/typescript/documentation/assets.ts
// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0

// DOCS_BLOCK_START:assets
import { Construct } from "constructs";
import { TerraformStack } from "cdktf";
import { TerraformAsset, AssetType } from "cdktf";
import { AwsProvider } from "./.gen/providers/aws/provider";
import { S3Bucket } from "./.gen/providers/aws/s3-bucket";
import { S3BucketObject } from "./.gen/providers/aws/s3-bucket-object";
import * as path from "path";

export class AssetsStack extends TerraformStack {
  constructor(scope: Construct, name: string) {
    super(scope, name);

    new AwsProvider(this, "aws", {
      region: "us-west-2",
    });

    const bucket = new S3Bucket(this, "bucket", {
      bucket: "demo",
    });

    const asset = new TerraformAsset(this, "lambda-asset", {
      path: path.resolve(__dirname, "lambda"),
      type: AssetType.ARCHIVE, // if left empty it infers directory and file
    });

    new S3BucketObject(this, "lambda-archive", {
      bucket: bucket.bucket,
      key: asset.fileName,
      source: asset.path, // returns a posix path
    });
  }
}
// DOCS_BLOCK_END:assets
